### PR TITLE
Extra debug information in HPA events

### PR DESCRIPTION
**Pull request by mwielgus**
Fixes: #29004

### CODE REVIEW

File: pkg/controller/podautoscaler/metrics/metrics_client.go [hunk 1]
```diff
@@ -172,7 +172,21 @@ func (h *HeapsterMetricsClient) getCpuUtilizationForPods(namespace string, selec
 	}
 
 	if len(metrics) != len(podNames) {
-		return 0, time.Time{}, fmt.Errorf("metrics obtained for %d/%d of pods", len(metrics), len(podNames))
+		present := make(map[string]struct{})
```
Comment (jszczepkowski) [inline pkg/controller/podautoscaler/metrics/metrics_client.go]: please use sets.String from "k8s.io/kubernetes/pkg/util/sets"

File: pkg/controller/podautoscaler/metrics/metrics_client_test.go [hunk 2]
```diff
@@ -184,10 +184,12 @@ func buildPod(namespace, podName string, podLabels map[string]string, phase api.
 }
 
 func (tc *testCase) verifyResults(t *testing.T, val *float64, timestamp time.Time, err error) {
-	assert.Equal(t, tc.desiredError, err)
 	if tc.desiredError != nil {
+		assert.Error(t, err)
+		assert.Contains(t, fmt.Sprintf("%v", err), fmt.Sprintf("%v", tc.desiredError))
 		return
 	}
+	assert.NoError(t, err)
```
Comment (jszczepkowski) [inline pkg/controller/podautoscaler/metrics/metrics_client_test.go]: Please add test case for hint (add case w/ one pod missing and set desired error to contain the missing pod name)

Comment (mwielgus) [inline pkg/controller/podautoscaler/metrics/metrics_client_test.go]: Done.

**Comment by jszczepkowski** [comment]
LGTM

**Pull request merged**